### ISSUE TITLE
Initial installation fails with: "schema "zulip" does not exist"

### ISSUE THREAD

**Issue by maltokyo** [zulip/docker-zulip#230]
Installing a fresh install, first time on a new Debian Buster VPS with 32GB RAM.

Everything goes well until the end of the installation.  But for some reason, the zulip container keeps shutting down.    Would someone lend me some hints on what I need to do differently, or what info I need to share with you to solve it?

Here is my docker-compose (yes I am intentionally switching https off):

```
version: '2'
services:
  zulipmyorg_postgresql:
    image: 'zulip/zulip-postgresql:10'
    container_name: zulipmyorg_postgresql
    environment:
      POSTGRES_DB: 'zulip'
      POSTGRES_USER: 'zulip'
      # Note that you need to do a manual `ALTER ROLE` query if you
      # change this on a system after booting the postgres container
      # the first time on a host.  Instructions are available in README.md.
      POSTGRES_PASSWORD: 'removed0001'
    volumes:
      - '/var/lib/docker/volumes/zulipmyorg/_data_postgresql/data:/var/lib/postgresql/data:rw'
  zulipmyorg_memcached:
    image: 'quay.io/sameersbn/memcached:latest'
    container_name: zulipmyorg_memcached
    restart: always
  zulipmyorg_rabbitmq:
    image: 'rabbitmq:3.7.7'
    container_name: zulipmyorg_rabbitmq
    hostname: zulipmyorg_rabbitmq
    restart: always
    environment:
        RABBITMQ_DEFAULT_USER: 'zulip'
        RABBITMQ_DEFAULT_PASS: 'removed0002'
    volumes:
      - '/var/lib/docker/volumes/zulipmyorg/_data_rabbitmq:/var/lib/rabbitmq:rw'
  zulipmyorg_redis:
    image: 'quay.io/sameersbn/redis:latest'
    container_name: zulipmyorg_redis
    volumes:
      - '/var/lib/docker/volumes/zulipmyorg/_data_redis:/var/lib/redis:rw'
  zulipmyorg_zulip:
    image: 'zulip/docker-zulip:2.1.1-0'
    container_name: zulipmyorg_zulip
    build:
      context: .
      args:
        # Change these if you want to build zulip from a different repo/branch
        ZULIP_GIT_URL: https://github.com/zulip/zulip.git
        ZULIP_GIT_REF: 2.1.1
        # Set this up if you plan to use your own CA certificate bundle for building
        # CUSTOM_CA_CERTIFICATES:
    # zmalmark enable the ports lines only if not behind reverse proxy
    #ports:
      #- '80:80'
      #- '443:443'
    environment:
      FORCE_FIRST_START_INIT: 'True'
      DB_HOST: 'zulipmyorg_postgresql'
      DB_HOST_PORT: '5432'
      DB_USER: 'zulip'
      DISABLE_HTTPS: 'True' # zmalmark from here: https://github.com/zulip/docker-zulip/issues/137
      SSL_CERTIFICATE_GENERATION: 'self-signed'
      SETTING_MEMCACHED_LOCATION: 'zulipmyorg_memcached:11211'
      SETTING_RABBITMQ_HOST: 'zulipmyorg_rabbitmq'
      SETTING_REDIS_HOST: 'zulipmyorg_redis'
      SECRETS_email_password: 'avdwlxxmxfilbfns'
      # These should match POSTGRES_PASSWORD and RABBITMQ_DEFAULT_PASS.
      SECRETS_rabbitmq_password: 'removed0002'
      SECRETS_postgres_password: 'removed0001'
      SECRETS_secret_key: 'removed0003'
      SETTING_EXTERNAL_HOST: 'z.example.com'
      SETTING_ZULIP_ADMINISTRATOR: 'myemail@gmail.com'
      SETTING_EMAIL_HOST: 'smtp.gmail.com'  # e.g. smtp.example.com
      SETTING_EMAIL_HOST_USER: 'myemail@gmail.com'
      SETTING_EMAIL_PORT: '587'
      # It seems that the email server needs to use ssl or tls and can't be used without it
      SETTING_EMAIL_USE_SSL: 'False'
      SETTING_EMAIL_USE_TLS: 'True'
      ZULIP_AUTH_BACKENDS: 'EmailAuthBackend'
      # Uncomment this when configuring the mobile push notifications service
      SETTING_PUSH_NOTIFICATION_BOUNCER_URL: 'https://push.zulipchat.com'
      # zmalmark - I could not get these to work to turn on manual settings as per here "Manual Config" so trying to do ALL with env https://github.com/zulip/docker-zulip#running-a-zulip-server-with-docker-compose
      #MANUAL_CONFIGURATION: 'True'
      #LINK_SETTINGS_TO_DATA: 'True'
      # zmalmark for nginx reverse proxy

    volumes:
      - '/var/lib/docker/volumes/zulipmyorg/_data_zulip:/data:rw'
    ulimits:
      nofile:
        soft: 40000
        hard: 50000
# zmalmark - add this for reverse proxy network
networks:
    default:
       external:
         name: webproxy
```

Below is the output of `docker logs` for the affected container, no errors until here:
```
Zulip configuration succeeded.
Auto backup enabled.
=== End Initial Configuration Phase ===
=== Begin Bootstrap Phase ===
Waiting for database server to allow connections ...
Executing Zulip first start init ...
+++ readlink -f /home/zulip/deployments/current/scripts/setup/initialize-database
++ dirname /home/zulip/deployments/2020-01-04-01-13-16/scripts/setup/initialize-database
+ THIS_DIR=/home/zulip/deployments/2020-01-04-01-13-16/scripts/setup
+ cd /home/zulip/deployments/2020-01-04-01-13-16/scripts/setup/../..
+ ./manage.py checkconfig
+ ./manage.py migrate --noinput
Operations to perform:
  Apply all migrations: analytics, auth, confirmation, contenttypes, otp_static, otp_totp, sessions, social_django, two_factor, zerver
Running migrations:
Traceback (most recent call last):
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 62, in execute
    return self.cursor.execute(sql)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zerver/lib/db.py", line 31, in execute
    return wrapper_execute(self, super().execute, query, vars)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zerver/lib/db.py", line 18, in wrapper_execute
    return action(sql, params)
psycopg2.errors.InvalidSchemaName: schema "zulip" does not exist


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./manage.py", line 46, in <module>
    execute_from_command_line(sys.argv)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/migration.py", line 129, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/operations/special.py", line 108, in database_forwards
    self._run_sql(schema_editor, self.sql)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/operations/special.py", line 133, in _run_sql
    schema_editor.execute(statement, params=None)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/base/schema.py", line 136, in execute
    cursor.execute(sql, params)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 62, in execute
    return self.cursor.execute(sql)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zerver/lib/db.py", line 31, in execute
    return wrapper_execute(self, super().execute, query, vars)
  File "/home/zulip/deployments/2020-01-04-01-13-16/zerver/lib/db.py", line 18, in wrapper_execute
    return action(sql, params)
django.db.utils.ProgrammingError: schema "zulip" does not exist

  Applying zerver.0001_initial...Zulip first start database initi failed in "initialize-database" exit code 1. Exiting.
```

**Comment by timabbott** [comment]
I haven't seen this exception before.

The issue appears to be that the postgres container doesn't have the `zulip` user and schema created; that should happen on first boot of the `postgres` container.  Have you perhaps fiddled around with what username you're using for postgres and not cleared the postgres container's storage in between fiddles?

@andersk FYI in case this could be caused by our work on switching the postgres container we're using.

**Comment by maltokyo** [comment]
Hi @timabbott - Thank you for response!

Its a new install, so I deleted all content from volumes and all containers and started from sratch.

So here are the logs, with a different error:

`# docker logs zulipmyorg_postgresql`
```
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data ... ok
creating subdirectories ... ok
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default timezone ... UTC
selecting dynamic shared memory implementation ... posix
creating configuration files ... ok
running bootstrap script ... ok
sh: locale: not found
2020-01-04 03:36:11.258 UTC [31] WARNING:  no usable system locales were found
performing post-bootstrap initialization ... ok
syncing data to disk ... ok

Success. You can now start the database server using:

    pg_ctl -D /var/lib/postgresql/data -l logfile start


WARNING: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
waiting for server to start....2020-01-04 03:36:12.204 UTC [36] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2020-01-04 03:36:12.232 UTC [37] LOG:  database system was shut down at 2020-01-04 03:36:11 UTC
2020-01-04 03:36:12.241 UTC [36] LOG:  database system is ready to accept connections
 done
server started
CREATE DATABASE


/usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/zulip-create-db.sql
2020-01-04 03:36:12.594 UTC [47] ERROR:  role "zulip" already exists
2020-01-04 03:36:12.594 UTC [47] STATEMENT:  CREATE USER zulip;
psql:/docker-entrypoint-initdb.d/zulip-create-db.sql:1: ERROR:  role "zulip" already exists
```


`# docker logs zulipmyorg_zulip`
```
=== Begin Initial Configuration Phase ===
Preparing and linking the uploads folder ...
Prepared and linked the uploads directory.
Executing nginx configuration ...
Disabling https in nginx.
Notice: Compiled catalog for 7379561c6c63 in environment production in 2.47 seconds
Notice: /Stage[main]/Zulip::Base/File[/etc/zulip/settings.py]/ensure: created
Notice: /Stage[main]/Zulip::Base/File[/etc/zulip/zulip-secrets.conf]/ensure: created
Notice: /Stage[main]/Zulip::Apt_repository/Exec[setup_apt_repo]/returns: executed successfully
Notice: /Stage[main]/Zulip::App_frontend/File[/etc/nginx/sites-available/zulip-enterprise]/content: content changed '{md5}bbc2679b3e2e3e3152d42d1a03056b21' to '{md5}99c6de0d02bfac9e6fe1e7b936d0ea9c'
Notice: /Stage[main]/Zulip::Nginx/Service[nginx]: Triggered 'refresh' from 1 event
Notice: Applied catalog in 1.72 seconds
Nginx configuration succeeded.
Generating self-signed certificates ...
+ is_redhat=false
+ '[' -e /etc/redhat-release ']'
+ SSLDIR=/etc/ssl
+ KEYFILE=/etc/ssl/private/zulip.key
+ CERTFILE=/etc/ssl/certs/zulip.combined-chain.crt
+ '[' -n '' ']'
+ '[' -z '' ']'
+ '[' -e /etc/ssl/private/zulip.key ']'
+ '[' -e /etc/ssl/certs/zulip.combined-chain.crt ']'
+ rm -f /etc/ssl/private/zulip.key /etc/ssl/certs/zulip.combined-chain.crt
++ mktemp
+ config=/tmp/tmp.Wn5keNQDgj
+ trap 'rm -f "$config"' EXIT
+ cat
+ '[' false = true ']'
+ apt-get install -y openssl
Reading package lists...
Building dependency tree...
Reading state information...
openssl is already the newest version (1.1.1-1ubuntu2.1~18.04.5).
openssl set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+ openssl req -new -x509 -config /tmp/tmp.Wn5keNQDgj -days 3650 -nodes -sha256 -out /etc/ssl/certs/zulip.combined-chain.crt -keyout /etc/ssl/private/zulip.key
Generating a RSA private key
..........................+++++
..................................................+++++
writing new private key to '/etc/ssl/private/zulip.key'
-----
Cannot write random bytes:
139936652689856:error:2407007A:random number generator:RAND_write_file:Not a regular file:../crypto/rand/randfile.c:163:Filename=/dev/urandom
+ chmod 644 /etc/ssl/certs/zulip.combined-chain.crt
+ chmod 640 /etc/ssl/private/zulip.key
+ rm -f /tmp/tmp.Wn5keNQDgj
Self-signed certificate generation succeeded.
Certificates configuration succeeded.
Setting database configuration ...
Setting key "REMOTE_POSTGRES_HOST", type "string" in file "/etc/zulip/settings.py".
Setting key "REMOTE_POSTGRES_SSLMODE", type "string" in file "/etc/zulip/settings.py".
Database configuration succeeded.
Setting Zulip secrets ...
Generating Zulip secrets ...
Generated Camo config file /etc/default/camo
Generated new secrets in /etc/zulip/zulip-secrets.conf.
Secrets generation succeeded.
Zulip secrets configuration succeeded.
Activating authentication backends ...
Setting key "AUTHENTICATION_BACKENDS", type "array" in file "/etc/zulip/settings.py".
Adding authentication backend "EmailAuthBackend".
Authentication backend activation succeeded.
Executing Zulip configuration ...
Setting key "EMAIL_HOST", type "string" in file "/etc/zulip/settings.py".
Setting key "EMAIL_HOST_USER", type "string" in file "/etc/zulip/settings.py".
Setting key "EMAIL_PORT", type "integer" in file "/etc/zulip/settings.py".
Setting key "EMAIL_USE_SSL", type "bool" in file "/etc/zulip/settings.py".
Setting key "EMAIL_USE_TLS", type "bool" in file "/etc/zulip/settings.py".
Setting key "EXTERNAL_HOST", type "string" in file "/etc/zulip/settings.py".
Setting key "MEMCACHED_LOCATION", type "string" in file "/etc/zulip/settings.py".
Setting key "PUSH_NOTIFICATION_BOUNCER_URL", type "string" in file "/etc/zulip/settings.py".
Setting key "RABBITMQ_HOST", type "string" in file "/etc/zulip/settings.py".
Setting key "RABBITMQ_USER", type "string" in file "/etc/zulip/settings.py".
Setting key "RATE_LIMITING", type "bool" in file "/etc/zulip/settings.py".
Setting key "REDIS_HOST", type "string" in file "/etc/zulip/settings.py".
Setting key "REDIS_PORT", type "integer" in file "/etc/zulip/settings.py".
Setting key "ZULIP_ADMINISTRATOR", type "string" in file "/etc/zulip/settings.py".
Zulip configuration succeeded.
Auto backup enabled.
=== End Initial Configuration Phase ===
=== Begin Bootstrap Phase ===
Waiting for database server to allow connections ...
............................................................Could not connect to database server. Exiting.
```

**Comment by maltokyo** [comment]
Also, when I now try to restart the container, I get the same error as in the first post above.

**Comment by maltokyo** [comment]
@timabbott & @andersk I tried again over the weekend to work out what to do to deploy this, I cannot make it work.  Perhaps my docker skills are not where they should be to troubleshoot.  Any advice or help would be good!  Thanks

**Comment by R4WBIT** [comment]
I'm currently having the same issue, and I've installed Zulip on Docker before, so something seems to be odd in the installation-process.

I'm testing from a clean server too, and am available for any troubleshooting needed.

**Comment by R4WBIT** [comment]
No, I also get the 'schema: "zulip" does not exist' error on startup. @timabbott 

```
Traceback (most recent call last):
  File "./manage.py", line 46, in <module>
    execute_from_command_line(sys.argv)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/migration.py", line 129, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/operations/special.py", line 108, in database_forwards
    self._run_sql(schema_editor, self.sql)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/migrations/operations/special.py", line 133, in _run_sql
    schema_editor.execute(statement, params=None)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/base/schema.py", line 136, in execute
    cursor.execute(sql, params)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zulip-py3-venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 62, in execute
    return self.cursor.execute(sql)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zerver/lib/db.py", line 31, in execute
    return wrapper_execute(self, super().execute, query, vars)
  File "/home/zulip/deployments/2019-12-14-03-04-44/zerver/lib/db.py", line 18, in wrapper_execute
    return action(sql, params)
django.db.utils.ProgrammingError: schema "zulip" does not exist
```

**Comment by Fr3E** [comment]
Have same problem, even when composing from 2.0.8-0 tag

**Comment by dangarthwaite** [comment]
Ditto.

**Comment by timabbott** [comment]
So, @andersk and I have been update to reproduce the issue, and I think it can be caused by removing the state of some containers but not others in a Docker cluster.  Can one of the folks seeing this try testing with a fresh clone of `docker-zulip` with a fresh state directory?

**Comment by dangarthwaite** [comment]
I can.  That's what this was - this was the first time trying zulip-docker.
I can also reproduce with:
docker-compose stop 
docker-compose rm
rm -fr /opt/docker/zulip/
docker-compose up --build --pull

**Comment by KoltesDigital** [comment]
I'm having this problem as well. User and database exist, but the schema does not. Running psql on the postgres container and manually adding the schema is enough for the server to start, but of course it's not an automated solution.

**Comment by maltokyo** [comment]
@timabbott do you need to login temporarily to a server to test this? I'm on Debian Buster is it helps. What is needed from us. Can we provide logs, anything?

**Comment by timabbott** [comment]
@maltokyo we're investigating this; it's clearly an issue a lot of people are having in the last few days.

**Comment by andersk** [comment]
It seems like the Postgres image changed at some point such that it creates the `zulip` user and database before calling `/docker-entrypoint-initdb.d/zulip-create-db.sql`, which then fails. I’m having trouble pinpointing exactly when this change occurred. However, zulip/zulip#13683 should fix it.

**Comment by timabbott** [comment]
That's merged; I think we're ready to release 2.1.2 from the 2.1.x branch either tonight or early tomorrow (And then a Docker release with the fix).

**Comment by maltokyo** [comment]
@timabbott thank you. Please let us know when it's available to test. I can this weekend.
